File: commands/branch-commands/createBranch.js
import simpleGit from "simple-git";

const git = simpleGit();

export default {
  command: "create-branch [branchName]",
  describe: "Create a new branch",
  builder: (yargs) => {
    yargs.positional("branchName", {
      describe: "Name of the new branch",
      type: "string",
      demandOption: true,
    });
  },
  handler: async (argv) => {
    try {
      await git.checkoutLocalBranch(argv.branchName);
      console.log(`Branch "${argv.branchName}" created successfully`);
    } catch (error) {
      console.error("Error creating branch:", error.message);
    }
  },
};

File: cmd/createBranch.go
package cmd

import (
	"fmt"
	"log"
	"os/exec"

	"github.com/spf13/cobra"
)

var CreateBranchCmd = &cobra.Command{
	Use: "branch",
	Short: "Create a new branch",
	Run:  func(cmd *cobra.Command, args []string) {
		if len(args) < 1 {
			fmt.Println("Error: You must provide a branch name")
			return
		}

		branchName := args[0]

		err := exec.Command("git", "branch", branchName).Run()
		if err != nil {
			log.Fatalf("Error creating branch '%s': %v", branchName, err)
		}
		fmt.Printf("Branch '%s' created successfully.\n", branchName)
	},
}

func init() {
	//No flags
}
